app/src/main/java/pl/futurecollars/invoicing/db/sql/SqlInvoiceRepository.java

```diff
@@ -0,0 +1,288 @@
+                    + "(select seller from invoice where id = '" + id + "')",
+                    updatedInvoice.getSeller().getTaxIdentifier(),
+                    updatedInvoice.getSeller().getName(),
+                    updatedInvoice.getSeller().getPensionInsurance(),
+                    updatedInvoice.getSeller().getHealthInsurance()
+            );
+
+            jdbcTemplate.update(
+                    "update company set tax_identifier = ?, name = ?, health_insurance = ?, pension_insurance = ? where id = "
+                        + "(select buyer from invoice where id = '" + id + "')",
+                    updatedInvoice.getBuyer().getTaxIdentifier(),
+                    updatedInvoice.getBuyer().getName(),
+                    updatedInvoice.getBuyer().getPensionInsurance(),
+                    updatedInvoice.getBuyer().getHealthInsurance()
+            );
+
+            jdbcTemplate.update(
+                    "update address set city = ?, postal_code = ?, street_name = ?, street_number = ? where company_id = "
+                        + "(select seller from invoice where id = '" + id + "')",
+                    updatedInvoice.getSeller().getAddress().getCity(),
+                    updatedInvoice.getSeller().getAddress().getPostalCode(),
+                    updatedInvoice.getSeller().getAddress().getStreetName(),
+                    updatedInvoice.getSeller().getAddress().getStreetNumber()
+            );
+
+            jdbcTemplate.update(
+                    "update address set city = ?, postal_code = ?, street_name = ?, street_number = ? where company_id = "
+                        + "(select buyer from invoice where id = '" + id + "')",
+                    updatedInvoice.getBuyer().getAddress().getCity(),
+                    updatedInvoice.getBuyer().getAddress().getPostalCode(),
+                    updatedInvoice.getBuyer().getAddress().getStreetName(),
+                    updatedInvoice.getBuyer().getAddress().getStreetNumber()
+            );
+
+            jdbcTemplate.update("update invoice set date_at = ?, invoice_number = ? where id = '" + id + "';",
+                updatedInvoice.getDateAt(),
+                updatedInvoice.getNumber()
+            );
+
+            jdbcTemplate.execute("delete from invoice_entry where invoice = '" + id + "'");
+
+            updatedInvoice.getEntries().forEach(entry -> {
+                jdbcTemplate.update("update invoice_entry set description = ?, price = ?, vat_value = ?, vat_rate = ?, invoice = ?",
+                    entry.getDescription(),
+                    entry.getPrice(),
+                    entry.getVatValue(),
+                    vatToId.get(entry.getVatRate()),
+                    id
+                );
+            });
+            return updatedInvoice;
+        } else {
+            throw new IllegalArgumentException("Invoice with id: " + id + " doesn't exist.");
```

Comment:
NoSuchElementException would be better https://docs.oracle.com/javase/8/docs/api/java/util/NoSuchElementException.html

Reply:
Done.